src/java/picard/analysis/directed/InsertSizeMetricsCollector.java

```diff
@@ -121,7 +133,13 @@ public void addMetricsToFile(final MetricsFile<InsertSizeMetrics,Integer> file)
                 final double total = Histogram.getCount();
 
                 // Only include a category if it has a sufficient percentage of the data in it
-                if( total > totalInserts * minimumPct ) {
+                if( total >= totalInserts * minimumPct ) {
+                    // in the case that minimumPct == 0 and total == 0, just make an empty histogram with zero values for the entire range
+                    if (Histogram.isEmpty()) {
+                        for (int key = minKey; key <= maxKey; key++) Histogram.increment(key, 0);
+                        if (Histogram.isEmpty()) throw new PicardException("BUG: minKey=" + minKey + " maxKey=" + maxKey);
```

Comment:
Is this to catch the case where ```minKey > maxKey```? If so, the error message should be more specific.